src/revparse.c

```diff
@@ -899,9 +899,17 @@ int git_revparse(
 			rstr++;
 		}
 
-		error = git_revparse_single(&revspec->from, repo, lstr);
-		if (!error)
-			error = git_revparse_single(&revspec->to, repo, rstr);
+		error = git_revparse_single(
+			&revspec->from,
+			repo,
+			*lstr == '\0' ? "HEAD" : lstr);
+
+		if (!error) {
+			error = git_revparse_single(
+				&revspec->to,
+				repo,
+				*rstr == '\0' ? "HEAD" : rstr);
+		}
```

Comment:
I hadn't thought about it much because `man gitrevisions` seems to explicitly say that the empty ranges are allowed. ("Note that .. would mean HEAD..HEAD which is an empty range that is both reachable and unreachable from HEAD.") However, on inspection, git.git treats `..` as a special case [on purpose](https://github.com/git/git/commit/003c84f6d2b9e9c4d5bbf5262cae994bac7190cb) so that it's always handled as a path for potentially ambiguous commands like `git log ..`. If the argument is disambiguated, `git rev-parse` gives an error:

```
$ git rev-parse .. --
fatal: bad revision '..'
```

I think it makes sense to reject `..`, since trying to parse an argument as a revision and then as a path is a pretty common git CLI idiom. Continuing to allow `...` seems a bit weird, but it should probably be accepted since git.git supports it.